src/main/java/com/uber/jenkins/phabricator/PhabricatorNotifier.java

```diff
@@ -128,42 +126,29 @@ public final boolean perform(final AbstractBuild<?, ?> build, final Launcher lau
 
         String phid = environment.get(PhabricatorPlugin.PHID_FIELD);
 
-        boolean runHarbormaster = phid != null && !"".equals(phid);
+        boolean runHarbormaster = !CommonUtils.isBlank(phid);
         boolean harbormasterSuccess = false;
 
-        String comment = null;
+        CommentBuilder commenter = new CommentBuilder(logger, build.getResult(), coverage, environment.get("BUILD_URL"));
 
-        if (coverage != null) {
-            Ratio lineCoverage = coverage.getCoverage(CoverageMetric.LINE);
-            if (lineCoverage == null) {
-                logger.info("uberalls", "no line coverage found, skipping...");
+        // First add in info about the change in coverage, if applicable
+        if (commenter.hasCoverageAvailable()) {
+            if (uberallsConfigured) {
+                commenter.processParentCoverage(uberalls.getParentCoverage(diff), diff.getBranch());
             } else {
-                if (uberallsConfigured) {
-                    comment = getCoverageComment(lineCoverage, uberalls, diff, logger.getStream(), environment.get("BUILD_URL"));
-                } else {
-                    logger.info("uberalls", "no backend configured, skipping...");
-                }
+                logger.info("uberalls", "no backend configured, skipping...");
             }
+        } else {
+            logger.info("uberalls", "no line coverage found, skipping...");
         }
 
         if (build.getResult().isBetterOrEqualTo(Result.SUCCESS)) {
```

Comment:
Can we just remove this entire thing and set the flag when the variable is instantiated?